src/plots/cartesian/layout_attributes.js

```diff
@@ -465,6 +465,36 @@ module.exports = {
             'Only has an effect if `anchor` is set to *free*.'
         ].join(' ')
     },
+    categorymode: {
+        valType: 'enumerated',
+        values: [
+            'trace', 'category ascending', 'category descending', 'array'
```

Comment:
@cldougl This is a good point too and somewhat hard to reconcile with the point from @mdtusz. You guys don't want to know my original version which was something like `domain-alphanumerical-ascending` (to allow for numerical etc. sorting and codomain/range based sorting). So it's a conflict, there's value in brevity and clarity too.